src/arm/debugger/debugger.c

```diff
@@ -126,8 +118,9 @@ static bool ARMDebuggerUpdateStackTraceInternal(struct mDebuggerPlatform* d, uin
 			bool isBranch = ARMInstructionIsBranch(info.mnemonic);
 			int reg = (isBranch ? info.op1.reg : info.op2.reg);
 			destAddress = cpu->gprs[reg];
-			if ((info.operandFormat & ARM_OPERAND_MEMORY_2) && (info.branchType & ARM_BRANCH_INDIRECT)) {
-				destAddress = cpu->memory.load32(cpu, destAddress, NULL);
+			if (!isBranch && info.branchType & ARM_BRANCH_INDIRECT && info.op1.reg == ARM_PC && info.operandFormat & ARM_OPERAND_MEMORY_2) {
```

Comment:
Wouldn't the mixing of `&` and `&&` generate a warning?

Reply:
You'd think, but apparently not. I'll put parens on it anyway.